### PR TITLE
Feature/area detail and delete

### PR DESCRIPTION
**Pull request by Bluesmile82**
-Add detail area page
-Add image and button with delete functionality
-Delete only if we have connection. Else show an alert
-Go back and sync when we delete the area

### CODE REVIEW

File: app/components/map/index.js [hunk 1]
```diff
@@ -181,6 +234,17 @@ class Map extends Component {
...
 
+  createReport = () => {
+    const { lastPosition } = this.state;
+    const form = `New-report-${Math.floor(Math.random() * 1000)}`;
+    let latLng = '0,0';
+    if (lastPosition) {
+      latLng = `${lastPosition.latitude},${lastPosition.longitude}`;
+    }
+    this.props.createReport(form, latLng);
+    this.props.navigate('NewReport', { form });
```
Comment (geriux) [inline app/components/map/index.js]: Lets change it to `back` instead so we can go back to the home page from the settings

File: app/components/settings/area-detail/index.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        I18n.t('commonText.connectionRequiredTitle'),
+        I18n.t('commonText.connectionRequired'),
+        [{ text: 'OK' }],
+        { cancelable: false }
+      );
+    }
+  }
+
+  render() {
+    const imageUrl = this.props.imageUrl !== undefined ? this.props.imageUrl : null;
```
Comment (geriux) [inline app/components/settings/area-detail/index.js]: Lets ask @juancarlosalonso for a nice image place holder 😬

File: app/components/settings/area-detail/styles.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  },
+  content: {
+    paddingTop: 0,
+    paddingBottom: 30
+  },
+  buttonContainer: {
+    padding: 10
+  },
+  imageContainer: {
+    backgroundColor: Theme.background.modal
```
Comment (geriux) [inline app/components/settings/area-detail/styles.js]: Lets center its childs for bigger screens

File: app/components/settings/area-list/index.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            >
+              <Image style={Theme.icon} source={nextIcon} />
+            </TouchableHighlight>
+          </View>
+        </TouchableHighlight>
+      )
+      })
+    return ( 
+      <View>
+        { this.props.syncing ?
```
Comment (geriux) [inline app/components/settings/area-list/index.js]: This is not showing up 😅 can you please check it?

File: app/components/settings/index.js [hunk 5]
```diff
@@ -89,47 +89,14 @@ class Settings extends Component {
...
-                        underlayColor="transparent"
-                      >
-                        <Image style={Theme.icon} source={nextIcon} />
-                      </TouchableHighlight>
-                    </View>
-                  </TouchableHighlight>
-                );
-              })
-              }
+              <AreaList areas={areas} areasImages={areasImages} navigate={navigate} syncing={syncing} />
```
Comment (geriux) [inline app/components/settings/index.js]: Lets make a container for this new `AreaList` component so we don't have to pass this data to it.

File: app/components/setup/draw-areas/index.js [hunk 6]
```diff
@@ -89,7 +90,8 @@ class DrawAreas extends Component {
     super(props);
     const intialCoords = this.props.country && this.props.country.centroid
       ? this.props.country.centroid.coordinates
-      : [Config.maps.lng, Config.maps.lat];
+      : [CONSTANTS.maps.lng, CONSTANTS.maps.lat];
+    console.log('initialcoords',intialCoords )
```
Comment (geriux) [inline app/components/setup/draw-areas/index.js]: Can we please remove this? 😁

**Comment by geriux** [comment]
And resolve conflicts please!

**Comment by geriux** [comment]
Also let's add the missing delete icon please 🙏

File: app/components/settings/area-detail/index.js (hunk 2 above)
Comment (Bluesmile82) [inline app/components/settings/area-detail/index.js]: We were trying to think about a case in which we would not have an image after the image API fix but we couldn't find any so I am not going to add the placeholder image right now

File: app/components/settings/area-list/index.js (hunk 4 above)
Comment (Bluesmile82) [inline app/components/settings/area-list/index.js]: Is showing for long delays. I can't find the specific doc but in most of frameworks you don't show the spinner if the call takes less than one second to load

**Comment by geriux** [comment]
Awesome!!!

**Pull request merged**